pkg/manager/member/tiflash_scaler.go

```diff
@@ -67,96 +69,123 @@ func (s *tiflashScaler) ScaleOut(meta metav1.Object, oldSet *apps.StatefulSet, n
 }
 
 func (s *tiflashScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
+	ns := meta.GetNamespace()
+	tcName := meta.GetName()
 	tc, ok := meta.(*v1alpha1.TidbCluster)
 	if !ok {
-		return nil
+		return fmt.Errorf("tiflashScaler.ScaleIn: failed to convert cluster %s/%s", meta.GetNamespace(), meta.GetName())
```

Comment:
Return error here will cause the TC to be resynced, however, what's the point of resync here if it's not a TC?

Reply:
Sorry I didn't notice that TiFlash ignore this error since it's determined an error in TiKV.

So the difference here is by design ?

Reply:
Actually, I think TiKV should also ignore this error.